R/Filter.R

```diff
@@ -47,6 +48,19 @@ autoplot.Filter = function(object, type = "boxplot", n = Inf, ...) { # nolint
   )
 }
 
+#' @rdname autoplot_filter
+#' @export
+autoplot.PipeOpFilter = function(object, type = "boxplot", n = Inf, ...) { # nolint
+  assert_true(!is.null(object$state))
+  filter = object$filter
+  prev_scores = filter$scores
+  on.exit({
```

Comment:
why  no `$clone()`?

Reply:
instead of the on.exit?

Reply:
I guess instead of `object$filter`?

How does `on.exit()` help here? First time I see it being used in a plotting scenario.

Reply:
PipeOpFilter stores a filter, but the filter does not contain the scores (threse are stored in the state of the pipeop). 
I assign the scores to the filter object and use the standard autoplot function for the filter object. Then I reset the fiter's scores to its previous scores (which should by NULL anyway) just to be sure

Reply:
@mllg what did you mean here?

Reply:
But why do we call `geom_bar` when `type = "boxplot"` ?

Reply:
I get that we could also do `filter = object$filter$clone()`, but why is it preferable to my `on.exit` ?